Source/WebCore/page/Navigation.cpp

```diff
@@ -122,14 +122,16 @@ void Navigation::initializeForNewWindow(std::optional<NavigationNavigationType>
     RefPtr currentItem = frame()->history().currentItem();
     if (!currentItem)
         return;
-    if (previousWindow) {
+    if (previousWindow && !frame()->isMainFrame()) {
```

Comment:
Could you elaborate on what is special about the mainFrame?

Reply:
I added a comment.